### PR TITLE
spatial/r2: implement Rotate

### PR DESCRIPTION
**Pull request by sbinet**
Fixes gonum/gonum#1513.

Please take a look.

### CODE REVIEW

File: spatial/r2/vector.go [hunk 1]
```diff
@@ -42,6 +42,11 @@ func (p Vec) Cross(q Vec) float64 {
 	return p.X*q.Y - p.Y*q.X
 }
 
+// Rotate returns a new vector, rotated by alpha around the provided vector.
```
Comment (kortschak) [inline spatial/r2/vector.go:45]:
```
// Rotate returns a new vector, rotated by alpha around the provided point, q.
```

Rotating around a vector in R2 feels odd.

**Review by kortschak — COMMENTED** [review]
It might be worth looking at the construction that gio uses for 2D transformations.

**Comment by sbinet** [comment]
Gio is using a more general type: `Affine2D` that not only handles rotations but also scaling, translations, etc...

- https://godoc.org/gioui.org/f32#Affine2D
- https://git.sr.ht/~eliasnaur/gio/tree/master/f32/affine.go#L12
- https://git.sr.ht/~eliasnaur/gio/tree/master/f32/affine.go#L51 (`Affine2D.Rotate(origin Point, angle float32) Affine2D`)
- https://git.sr.ht/~eliasnaur/gio/tree/master/f32/affine.go#L122

having used for implementing dashes and strokes for `Gio` and `vg-gio`, it's well thought out.
but I haven't had the use for such a complete affine transformation in my day-to-day "science-y" work.

**Comment by kortschak** [comment]
Yeah, fair enough.

**Review by kortschak — APPROVED** [review]

**Comment by sbinet** [comment]
do you know what this "Travis CI - Branch" item is?

it doesn't seem to ever go green (nor red).
the `r3-vec-rotate` PR had that field yellow for the whole Fri->Mon interval...

**Comment by kortschak** [comment]
It looks like the action that travis uses to run travis-ci.com is not aware of the situation when it doesn't have access to the branch. I can make that one not required, but you should be able to merge anyway with one extra click.

**Comment by sbinet** [comment]
yeah, that's what I did for `r3-vec-rotate` :)

**Pull request merged**